src/mbgl/style/conversion/transition_options.cpp

```diff
@@ -10,7 +10,10 @@ optional<TransitionOptions> Converter<TransitionOptions>::operator()(const Conve
         return {};
     }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
```

Comment:
Alternatively, we could have an empty `optional<TransitionOptions>` object, that could be returned in case of error, thus avoiding these pragma calls.